wallet/src/params.rs

```diff
@@ -43,6 +44,54 @@ impl Default for Params {
 
 #[derive(Clone)]
 pub struct NodeParams {
-    pub address: Addr<JsonRpcClient>,
+    /// The IP address and port of the node we are connecting to.
+    pub address: String,
+    /// Reference to the JSON-RPC client actor.
+    pub client: Addr<JsonRpcClient>,
+    /// A reference to the latest block that the node has consolidated into its block chain.
+    pub last_beacon: Arc<RwLock<CheckpointBeacon>>,
+    /// The name of the network in which the node is operating.
+    pub network: Arc<RwLock<Option<String>>>,
+    /// Timeout for JSON-RPC requests sent to the node.
     pub requests_timeout: Duration,
 }
+
+impl NodeParams {
+    /// Retrieve the `last_beacon` field.
+    /// This panics if the `RwLock` is poisoned.
+    pub fn get_last_beacon(&self) -> CheckpointBeacon {
+        let lock = (*self.last_beacon).read();
+        *lock.expect("Read locks should only fail if poisoned.")
+    }
+
+    /// Retrieve the `network` field.
+    /// This panics if the `RwLock` is poisoned.
+    pub fn get_network(&self) -> Option<String> {
+        let lock = (*self.network).read();
+        lock.expect("Read locks should only fail if poisoned.")
+            .clone()
+    }
+
+    /// Update the `last_beacon` field with the information of the latest block that the node has
+    /// consolidated into its block chain.
+    /// This is a best-effort method. It will silently do nothing if the write lock on `last_beacon`
+    /// cannot be acquired or if the new beacon looks older than the current one.  
+    pub fn update_last_beacon(&self, new_beacon: CheckpointBeacon) {
+        let lock = (*self.last_beacon).write();
+        if let Ok(mut beacon) = lock {
+            if new_beacon.checkpoint > beacon.checkpoint {
+                *beacon = new_beacon
+            }
+        }
+    }
+
+    /*
+    /// Update the `network` field in runtime, as it is unknown at boot.
+    /// This is a best-effort method. It will silently do nothing if the write lock on `network`.
+    pub fn update_network(&self, new_network: String) {
+        let lock = (*self.network).write();
+        if let Ok(mut network) = lock {
+            *network = Some(new_network)
+        }
+    }*/
```

Comment:
As explained in the PR comment, I'm still undecided on what to do with this.

We could try to guess the network from a few different places, e.g. last bytes of the bootstrap hash.

Of course we could ask the node through JSON-RPC. But the bottom line is that not even the node is aware of what's the "identifier" of the network it is operating on. 

So maybe I'd go with a combination of those two things: asking the node and let the node guess from the bootstrap hash.

Reply:
If we want to show the network as "testnet-7", "testnet-8", "mainnet", or even some custom names, then we can add a `network_name` field to `witnet.toml`, and add that to the status JSON-RPC method of the node. If we only want testnet/mainnet then I think there is a variable in the wallet like `testnet = true`, and the wallet will show an error when connecting to a node running on mainnet because the addresses have a different prefix (twit vs wit).

Reply:
Let's do the latter for the time being.